### PR TITLE
fix: spoof origin and referer

### PR DESCRIPTION
**Pull request by JFrankfurt**
## Description
this simplifies our interceptors and enables fetching quotes from the routing api by adding a referer spoof

### CODE REVIEW

File: cypress/support/e2e.ts [hunk 1]
```diff
@@ -89,27 +89,22 @@ Cypress.Commands.overwrite(
 )
 
 beforeEach(() => {
-  // Infura security policies are based on Origin headers.
-  // These are stripped by cypress because chromeWebSecurity === false; this adds them back in.
-  cy.intercept(/infura.io/, (res) => {
-    res.headers['origin'] = 'http://localhost:3000'
-    res.alias = res.body.method
-    res.continue()
+  cy.intercept('*', (req) => {
```
Comment (zzmp) [inline cypress/support/e2e.ts:92]: Add a comment above.

File: cypress/support/e2e.ts [hunk 2]
```diff
@@ -89,27 +89,22 @@ Cypress.Commands.overwrite(
...
+  // Infura security policies are based on Origin headers.
   // These are stripped by cypress because chromeWebSecurity === false; this adds them back in.
-  cy.intercept('https://api.uniswap.org/v1/graphql', (res) => {
-    res.headers['origin'] = 'https://app.uniswap.org'
-    res.continue()
-  })
-  cy.intercept('https://beta.api.uniswap.org/v1/graphql', (res) => {
-    res.headers['origin'] = 'https://app.uniswap.org'
-    res.continue()
+  cy.intercept(/infura.io/, (req) => {
```
Comment (zzmp) [inline cypress/support/e2e.ts:99]: Can these calls be chained?

File: cypress/support/e2e.ts [hunk 3]
```diff
@@ -89,27 +89,22 @@ Cypress.Commands.overwrite(
...
+    req.headers['referer'] = 'http://localhost:3000'
+    req.headers['origin'] = 'http://localhost:3000'
+    req.alias = req.body.method
+    req.continue()
   })
 
-  cy.intercept('https://api.uniswap.org/v1/amplitude-proxy', (res) => {
-    res.reply(JSON.stringify({}))
+  cy.intercept('https://api.uniswap.org/v1/amplitude-proxy', (req) => {
+    req.reply(JSON.stringify({}))
```
Comment (zzmp) [inline cypress/support/e2e.ts:107]: nit: Is there a way to do this such that we don't see pending amplitude POSTs in the console?
**This is out of scope**: I'm just wondering if you've looked into as part of this work.

**Review by zzmp — CHANGES_REQUESTED** [review]

File: cypress/support/e2e.ts [hunk 4]
```diff
@@ -89,28 +89,32 @@ Cypress.Commands.overwrite(
 )
 
 beforeEach(() => {
-  // Infura security policies are based on Origin headers.
-  // These are stripped by cypress because chromeWebSecurity === false; this adds them back in.
-  cy.intercept(/infura.io/, (res) => {
-    res.headers['origin'] = 'http://localhost:3000'
-    res.alias = res.body.method
-    res.continue()
+  // many of our api calls enforce these headers come from our app
```
Comment (zzmp) [inline cypress/support/e2e.ts]:
```suggestion
  // Many API calls enforce that requests come from our app, so we must mock Origin and Referer.
```

File: cypress/support/e2e.ts [hunk 5]
```diff
@@ -89,28 +89,32 @@ Cypress.Commands.overwrite(
...
+  cy.intercept('*', (req) => {
+    req.headers['referer'] = 'https://app.uniswap.org'
+    req.headers['origin'] = 'https://app.uniswap.org'
   })
+    // Infura security policies are based on Origin headers.
+    // These are stripped by cypress because chromeWebSecurity === false; this adds them back in.
```
Comment (zzmp) [inline cypress/support/e2e.ts:98]:
```suggestion
    // Infura uses a test endpoint, which allow-lists http://localhost:3000 instead.
```

File: cypress/support/e2e.ts [hunk 6]
```diff
@@ -89,28 +89,32 @@ Cypress.Commands.overwrite(
...
-  })
-  cy.intercept('https://beta.api.uniswap.org/v1/graphql', (res) => {
-    res.headers['origin'] = 'https://app.uniswap.org'
-    res.continue()
-  })
-
-  cy.intercept('https://api.uniswap.org/v1/amplitude-proxy', (res) => {
-    res.reply(JSON.stringify({}))
-  })
+    .intercept('https://api.uniswap.org/v1/amplitude-proxy', (req) => {
```
Comment (zzmp) [inline cypress/support/e2e.ts:106]:
```suggestion
    // Mock Amplitude responses to avoid analytics from tests. 
    .intercept('https://api.uniswap.org/v1/amplitude-proxy', (req) => {
```

**Review by zzmp — CHANGES_REQUESTED** [review]

**Review by zzmp — APPROVED** [review]

**Review by mikeki — APPROVED** [review]
LGTM

**Review by cartcrom — APPROVED** [review]
